baseband/vlbi_base/base.py

```diff
@@ -211,54 +212,55 @@ def __repr__(self):
 class VLBIStreamReaderBase(VLBIStreamBase):
 
     def __init__(self, fh_raw, header0, sample_shape, bps, complex_data,
-                 thread_ids, samples_per_frame, frames_per_second=None,
-                 sample_rate=None, squeeze=True):
+                 thread_ids, samples_per_frame, sample_rate=None,
+                 squeeze=True):
 
-        if frames_per_second is None and sample_rate is None:
+        if sample_rate is None:
             try:
-                frames_per_second = self._get_frame_rate(fh_raw,
-                                                         header0)
+                sample_rate = self._get_sample_rate(fh_raw, header0,
+                                                    samples_per_frame)
```

Comment:
Ah, now I see the possible pitfall: do all headers have `samples_per_frame` -- which is not true for Mark 5B.  How about just keeping this as `_get_frame_rate` -- after all, that is what the basic routine does if times fail it -- and then multiply with `samples_per_frame` here.